tuxemon/npc.py

```diff
@@ -778,6 +779,7 @@ def evolve_monster(self, old_monster: Monster, evolution: str) -> None:
         new_monster.current_hp = min(old_monster.current_hp, new_monster.hp)
         new_monster.moves = old_monster.moves
         new_monster.instance_id = old_monster.instance_id
+        new_monster.possible_genders = old_monster.possible_genders
```

Comment:
Just calculate the gender here and set it to an attribute called gender.

Reply:
Do you mean set_gender instead of this?

Reply:
I added
`self.gender = random.choice(list(results.possible_genders))`

Reply:
I didn't do other editing because I want to be sure about the following steps.